README
  The Small Arms Ballistics Handbook
  Joshua Weage <weage98@yahoo.com>
  Version 1.3.3, 2 July 2005

  This Handbook describes Small Arms Ballistics Version 1.3.3
  ______________________________________________________________________

  Table of Contents


  1. Introduction
     1.1 Changes

  2. Installation
     2.1 How to obtain Small Arms Ballistics
     2.2 Requirements
     2.3 Compilation and installation

  3. Usage
     3.1 Configuration Options
     3.2 Drag Models
        3.2.1 Available Drag Models
     3.3 Select Bullet
     3.4 Calculation Parameters
     3.5 Trajectory Table Output
     3.6 Calculate B.C.

  4. Information Sources
  5. Copyright


  ______________________________________________________________________

  [1m1.  Introduction[0m

  Small Arms Ballistics is, as the name says, a ballistics calculator
  for small arms fire.  It is a terminal application written using
  curses.

  Small Arms Ballistics is derived from smallball, which was written by
  Ole-Hj. Kristensen <okh@edeber.nta.no>.  I have pretty much rewritten
  the entire program, but smallball got me started.  All major
  trajectory calculations have been implemented, and a variety of bullet
  drag models are available.

  The program needs a color capable terminal, otherwise it won't run.


  [1m1.1.  Changes[0m

  See the ChangeLog.


  [1m2.  Installation[0m



  [1m2.1.  How to obtain Small Arms Ballistics[0m

  Small Arms Ballistics can be downloaded from www.ibiblio.org at the
  following location
  www.ibiblio.org/pub/linux/misc/saballistics-1.3.3.tar.gz.



  [1m2.2.  Requirements[0m

  This program requires the menu, form, and ncurses libaries and a color
  capable terminal emulator.  The program is written in ANSI C.


  [1m2.3.  Compilation and installation[0m

  In order to compile and install Small Arms Ballistics on your system,
  type the following in the base directory of the distribution:

  ./configure make make install



       % ./configure
       % make
       % make install



  Since Small Arms Ballistics uses

  autoconf


  you should have not trouble compiling it.  Should you run into prob-
  lems please report them to the the author at weage98@yahoo.com


  [1m3.  Usage[0m



  [1m3.1.  Configuration Options[0m


  +o  Unit System - English or Metric unit operation.

  +o  Drag Model - The drag function used to calculate projectile
     deceleration, see the DRAG MODELS section.

  +o  Maximum Flight Time - The projectile flight is calculated using an
     explicit time stepping integration routine.  This is the maximum
     simulated flight time.

  +o  Integration Points Per Second - The number of integration steps per
     second.  Values as low as 10 seem to produce good data.  1000 was
     used in the smallball program, so it is the default here.


  [1m3.2.  Drag Models[0m

  The deceleration of a bullet is calculated using standard drag models.
  These models were produced by firing a standard projectile at a range
  of velocities and recording its performance.  The ballistic
  coefficient relates any bullet to the standard model.  However, only
  projectiles shaped exactly the same as the standard will match the
  model's performance.

  For the best accuracy, recalculate the ballistic coefficient of the
  bullet using your chosen drag model.  Most manufacturers report the
  B.C. using the G1 model, so that is the best to start with.


  [1m3.2.1.  Available Drag Models[0m


  +o  Ingalls - Based on Krupp data of 1881

  +o  British - Based on British 1909 data

  +o  G1 - Standard drag model (similar to Ingalls)

  +o  G5 - Moderate boat tails - 7 degree 30' tail taper with 6.19
     caliber tangent nose ogive.

  +o  G6 - Flat base, spire points.  6.09 caliber secant nose ogive

  +o  G7 - "VLD" boat tail.  Long 7 degree 30' tail taper with 10 caliber
     tangent nose ogive.

  For .22LR use the G1 model.  The G1 is good for blunt nose, exposed
  lead bullets with a velocity less than 1400 ft/s.


  [1m3.3.  Select Bullet[0m

  I have included a datafile of bullets which was taken from PCB by Odd
  Harvard Skevik.  www.stud.ntnu.no/~oddske/ballistikk.html This file
  will be copied into the share directory.

  The first two lines of this file are skipped.  For bullet entries, the
  diameter and description must end at column 40 and are followed by the
  weight and the ballistic coefficient.  Feel free to add additional
  entries to this file, following the existing format.


  [1m3.4.  Calculation Parameters[0m


  +o  Ballistic Coefficient - Projectile ballistic coefficient.

  +o  Bullet Weight - Projectile weight in grams or grains.

  +o  Wind Velocity - Wind velocity at a 90 degree angle to projectile
     flight.

  +o  Muzzle Velocity - Initial velocity of projectile.

  +o  Bore-Sight Distance - Distance between the line of sight and the
     bore.

  +o  Zero Distance - Sight zero distance.

  +o  Firing Angle - Elevation angle of the shot.  The Path output in the
     trajectory table is calculated relative to this.

  +o  Starting Range - Starting distance listed in the trajectory output.

  +o  Ending Range - Last distance listed in the trajectory output.

  +o  Step Size - Range increment used for the trajectory output.


  [1m3.5.  Trajectory Table Output[0m


  +o  Range - The horizontal distance to the projectile location.  For
     calculations with a non-zero firing angle, the Path values are
     given for the range along the line of sight, not the horizontal
     distance.

  +o  Velocity - Absolute projectile velocity.

  +o  Energy - Projectile kinetic energy.

  +o  Mom. - Projectile momentum.

  +o  Path - Actual projectile flight path in relation to the line of
     sight.

  +o  Zero - Projectile flight path for a horizontal, zeroed shot.

  +o  Drop - Amount of drop if the projectile is fired perfectly
     horizontal.

  +o  Drift - Horizontal drift due to wind, along line of sight.

  +o  Time - Flight time of projectile.


  [1m3.6.  Calculate B.C.[0m

  This function calculates the projectile ballistic coefficient based on
  the input data and the selected drag model.


  [1m4.  Information Sources[0m

  There are several sites on the web that have good information on
  ballistics, these are the ones I recommend:


  +o  www.snipercountry.com/ballistics/index.html

  +o  www.lascruces.com:80/~jbm/


  [1m5.  Copyright[0m


  Small Arms Ballistics Copyright 2005 Joshua Weage, <weage98@yahoo.com>

  This program is free software; you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation; either version 2 of the License, or (at
  your option) any later version.

  This program is distributed in the hope that it will be useful, but
  WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
  General Public License for more details.

  You should have received a copy of the GNU General Public License
  along with this program; if not, write to the Free Software
  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.





ChangeLog
1.3.3
 2 July 2005 <weage98@yahoo.com>
 - redistribute as 1.3.2 tarball was not up to date

1.3.2
 30 June 2003 <weage98@yahoo.com>
 - distribution didn't include saballistics.lsm.in
 - Add code to detect dirent.h header properly
 - Error out of configure when missing required libs and headers
 - index.sgml.in now processed by autoconf
 - README must be copied from docs/en/index.txt after a build!
 - removed unneeded programs from configure.ac
 - Added conditional creation of documentation depending on
   the existance of the linuxdoc tool.
 - Added date substitutions into configure.ac.

29 Jun 2005: 1.3.1
 <weage98@yahoo.com>
 - Updated autotools config
 - Reorganized directory hierarchy
 - Add units header to saved files
 - Fixed problem with unit conversion and path calculation when
   writing to a file.
 - Added a few more comments to explain the calculation procedure.
 <geff@spamcop.net>
 - Fixed bug when writing out height of sight in english units.
   Height was reported as 12 * actual height in results_pad.c.

4 April 2004: 1.3.0
	- Moved the codebase to kdevelop
	- Cleaned up the code
	- Changed color scheme

5 Oct. 1999: v1.2
        - Added bullet database
        - Fixed bug in B.C. calculation.
        - Firing angle effects
        - Added flight time and integration points configuration

27 Sep. 1999: v1.1
        - Better docs
        - Additional trajectory table entries
        - Trajectory table save function
        - TAB, and ESC key recognition
        - "G" drag functions and descriptions

19 Sep. 1999: v1.0
        Released v1.0



src/messages.c
/***************************************************************************
                          messages.c  -  description
                             -------------------
    begin                : Tue Mar 30 2004
    copyright            : (C) 2004 by Joshua Weage
    email                : weage98@yahoo.com
 ***************************************************************************/

/***************************************************************************
 *                                                                         *
 *   This program is free software; you can redistribute it and/or modify  *
 *   it under the terms of the GNU General Public License as published by  *
 *   the Free Software Foundation; either version 2 of the License, or     *
 *   (at your option) any later version.                                   *
 *                                                                         *
 ***************************************************************************/

#ifdef HAVE_CONFIG_H
#include <config.h>
#endif

#include "messages.h"
#include "main.h"

#include <stdlib.h>

#ifdef ENGLISH_LANG
/* error strings */
char initstring[] = "Couldn't initialize screen";
char need_color_str[] = "This program requires a color terminal";

/* splashscreen strings */
char gpl1[] = "Small Arms Ballistics comes with ABSOLUTELY NO WARRANTY;";
char gpl2[] = "This is free software and you are welcome to distribute";
char gpl3[] = "it under certain conditions: see the file 'COPYING' for";
char gpl4[] = "details.";

char info1[] = "Small Arms Ballistics is " COPYRIGHT;
char info2[] = "It is derived from smallball, originally written by";
char info3[] = "Ole-Hj. Kristensen <ohk@edeber.nta.no>";

/* title window strings */
char name_str[] = "Small Arms Ballistics " VERSION;
char author_str[] = COPYRIGHT;
char usage_str[] = "Use the arrow keys to move, enter to select, `q' to exit.";

/* modify parameters strings */
char modusage_str[] = "Use the arrow keys to move, select DONE or `q' when finished.";

/* configuration strings */
char *units_str[] =
  {
    "ENGLISH",
    "METRIC"
  };

char *model_str[] =
  {
    "British",
    "Ingalls",
    "G1",
    "G2",
    "G5",
    "G6",
    "G7",
    "G8"
  };

char *model_desc_str[] =
  {
    "     British 1909 Model     ",
    "  Based on 1881 Krupp data  ",
    " G1 - Standard bullet model ",
    "        G2 - Unknown        ",
    "  G5 - Moderate boat tails  ",
    "G6 - Flat base, spire points",
    "    G7 - \"VLD\" boat tail    ",
    "        G8 - Unknown        "
  };

char *configuration_options[] =
  {
    "Unit System",
    "Drag Model",
    "Maximum Flight Time [s]",
    "Integration Points Per Second"
  };

/* calculate ballistic coefficient strings */
char *calc_bc_e[] =
  {
    "Initial Range [yds]",
    "Initial Velocity [ft/s]",
    "Final Range [yds]",
    "Final Velocity [ft/s]"
  };

char *calc_bc_m[] =
  {
    "Initial Range [m]",
    "Initial Velocity [m/s]",
    "Final Range [m]",
    "Final Velocity [m/s]"
  };

char *calc_bc_results = "The calculated B.C. is";

/* buttons */
char done_str[] = "  DONE";
char reset_str[] = " RESET";

/* main menu strings */
char *mainmenu_options[] =
  {
    "Configuration     ",
    "Select Bullet     ",
    "Modify Parameters ",
    "Compute Trajectory",
    "Calculate B.C.",
    "Quit Ballistic    "
  };

char *mainmenu_help[] =
  {
    "- Global Configuration Options ",
    "- Select Bullet From Database ",
    "- Modify Calculation Parameters ",
    "- Compute Trajectory ",
    "- Calculate Bullet B.C. ",
    "- Quit Small Arms Ballistics "
  };

/* select bullet strings */
char *bullet_str = "D [in]     Description                Weight [gr]     B.C.";

/* modify bullet paramaters strings */
char *calculation_parameters_e[] =
  {
    "Ballistic Coefficient",
    "Bullet Weight   [grains]",
    "Wind Velocity   [ft/s]",
    "Muzzle Velocity [ft/s]",
    "Bore-Sight Distance [in]",
    "Zero Distance   [yds]",
    "Firing Angle [deg]",
    "Starting Range  [yds]",
    "Ending Range    [yds]",
    "Step Size       [yds]"
  };

char *calculation_parameters_m[] =
  {
    "Ballistic Coefficient",
    "Bullet Weight   [grams]",
    "Wind Velocity   [m/s]",
    "Muzzle Velocity [m/s]",
    "Bore-Sight Distance [m]",
    "Zero Distance   [m]",
    "Firing Angle [deg]",
    "Starting Range  [m]",
    "Ending Range    [m]",
    "Step Size       [m]"
  };

/* calculation display strings */
                     /* 12345678901234567890123456789012345678901234567890 */
char calc_headings[] = "Range  Velocity  Energy   Mom.     Path    Zero    Drop  Drift   Time";
char calc_units_m[]  = " [m]     [m/s]    [kJ]  [kg.m/s]   [m]     [m]     [m]    [m]    [s]";
char calc_units_e[]  = "[yds]   [ft/s] [ft.lbf] [ft.lb/s]  [in]    [in]    [in]    [in]   [s]";
char calc_usage_str[] = "Use the arrow keys to move, `s' to save table, `q' to exit.";

/* save trajectory table strings */
char save_table_str[] = "Output Filename: ";
char opendir_err[] = "Couldn't open current directory";
char save_file_exists[] = "File already exists, overwrite (y/n)?";
char save_table_err[] = "Couldn't Open File, press RETURN";
#endif
